studygroups/views/facilitate.py

```diff
@@ -614,27 +619,46 @@ def form_valid(self, form):
         # add to team by token
         token = self.kwargs.get('token')
         team_from_token = self.get_team_from_token(token)
-        if team_from_token and invitation_confirmed:
-            current_membership_qs.delete()
-            new_membership = TeamMembership.objects.create(team=team_from_token, user=user, role=default_role)
+        if team_from_token:
+            if invitation_confirmed:
+                current_membership_qs.delete()
+                pending_invitations = TeamInvitation.objects.filter(email__iexact=user.email, responded_at__isnull=True)
+                pending_invitations.delete()
+                new_membership = TeamMembership.objects.create(team=team_from_token, user=user, role=default_role)
+                messages.success(self.request, _('Welcome to the team! You are now a part of {}.'.format(new_membership.team.name)))
+                return super(InvitationConfirm, self).form_valid(form)
+            else:
+                return super(InvitationConfirm, self).form_valid(form)
```

Comment:
@s-kennedy Wouldn't this mean that if you don't confirm the invitation, it just stays on your dash?

Reply:
That block is for the token case, when the user clicks on the invitation link, so there's no notification and no invitation. So if they don't confirm the invitation, nothing happens.

Reply:
nvm, you're right, i didn't read properly